homeassistant/components/cybro/binary_sensor.py

```diff
@@ -0,0 +1,118 @@
+
+from .const import (
+    AREA_SYSTEM,
+    DEVICE_DESCRIPTION,
+    DOMAIN,
+    MANUFACTURER,
+    MANUFACTURER_URL,
+)
+from .coordinator import CybroDataUpdateCoordinator
+from .models import CybroEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+    variable_name: str = "",
+) -> None:
+    """Set up a Cybro binary sensor based on a config entry."""
+    coordinator: CybroDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if variable_name == "":
+        var_prefix = f"c{coordinator.cybro.nad}."
+        dev_info = DeviceInfo(
+            identifiers={(DOMAIN, var_prefix)},
+            manufacturer=MANUFACTURER,
+            default_name=f"{var_prefix} Diagnosis",
+            suggested_area=AREA_SYSTEM,
+            model=f"{DEVICE_DESCRIPTION} controller",
+            configuration_url=MANUFACTURER_URL,
+        )
+        async_add_entities(
+            [
+                CybroUpdateBinarySensor(
+                    coordinator,
+                    var_prefix + "scan_overrun",
+                    attr_entity_category=EntityCategory.DIAGNOSTIC,
+                    attr_device_class=BinarySensorDeviceClass.PROBLEM,
+                    dev_info=dev_info,
+                ),
+                CybroUpdateBinarySensor(
+                    coordinator,
+                    var_prefix + "retentive_fail",
+                    attr_entity_category=EntityCategory.DIAGNOSTIC,
+                    attr_device_class=BinarySensorDeviceClass.PROBLEM,
+                    dev_info=dev_info,
+                ),
+                CybroUpdateBinarySensor(
+                    coordinator,
+                    var_prefix + "general_error",
+                    attr_entity_category=EntityCategory.DIAGNOSTIC,
+                    attr_device_class=BinarySensorDeviceClass.PROBLEM,
+                    dev_info=dev_info,
+                ),
+            ]
+        )
+        return
+    async_add_entities(
+        [
+            CybroUpdateBinarySensor(
+                coordinator,
+                variable_name,
+            )
+        ]
+    )
+
+
+class CybroUpdateBinarySensor(CybroEntity, BinarySensorEntity):
+    """An entity using CoordinatorEntity.
+
+    The CoordinatorEntity class provides:
+      should_poll
+      async_update
+      async_added_to_hass
+      available
+
+    """
+
+    def __init__(
+        self,
+        coordinator: CybroDataUpdateCoordinator,
+        var_name: str = "",
+        attr_entity_category: EntityCategory = None,
+        attr_device_class: BinarySensorDeviceClass = None,
+        dev_info: DeviceInfo = None,
+    ) -> None:
+        """Pass coordinator to CoordinatorEntity."""
+        super().__init__(coordinator=coordinator)
+        if var_name == "":
+            return
+        self._attr_name = var_name
+        self._attr_unique_id = var_name
```

Comment:
Is this `unique_id` unique?

Reply:
Yes, the `unique_id` is built as following:
Controller nad (=serial number of PLC) + name of variable in PLC which could only exist once